core/meta/src/TClass.cxx

```diff
@@ -1438,7 +1441,10 @@ void TClass::Init(const char *name, Version_t cversion,
          }
       }
       if (!fHasRootPcmInfo && gInterpreter->CheckClassInfo(fName, /* autoload = */ kTRUE)) {
-         gInterpreter->SetClassInfo(this);   // sets fClassInfo pointer
+         {
+            R__LOCKGUARD(gInterpreterMutex);
```

Comment:
> although it would be probably better to both auto parse and set class info all in one go by still taking the lock there.

Yes, this is what the already happens in LoadClassInfo.  I do not see how this lock does anything (since it is limited to a single statement (which turns around and takes the lock a second time) by the explicit scope).